application/models/Plugin.php

```diff
@@ -71,6 +71,9 @@ public function tableName()
      */
     public function setLoadError(array $error)
     {
+        if (App()->getConfig('debug') >= 2) {
```

Comment:
I am not sure about this mechanism.
This method, which is supposed to only save is somehow now deciding when the message should be thrown or not.

Why not making  method called "handleLoadError".
That method will decide if saving or throwing.
It will return something only when saving.

It may be too much, but I would move the disabling of the plugin inside that method as well.

Reply:
> I don't want to rename a public function.

I meant to create a new method, not rename a current one.
`handleLoadError` would call, in some cases, to `setPluginLoadError`

Reply:
Yes ! Great !